app/controllers/concerns/authentication/controller.rb

```diff
@@ -21,153 +20,185 @@ module Controller
-        options[:api_notice] ||= options[:notice]
-        options
-      end
+    def current_user
+      @current_user
+    end
 
-      def handle_unverified_user(options)
-        respond_to do |format|
-          format.any(:html, :mobile) do
-            flash[:notice] = options[:notice]
-            redirect_to options[:redirect]
-          end
-          format.json { render json: options[:api_notice], status: 401 }
-          format.xml  { render xml:  options[:api_notice], status: 401 }
+    def current_user?
+      current_user ? true : false
+    end
+
+    def set_current_user(user)
+      @current_user = user
+    end
+
+    def default_verify_user_options(options = {})
+      options[:redirect] ||= discussions_url
+      options[:notice] ||= "You don't have permission to do that!"
+      options[:api_notice] ||= options[:notice]
+      options
+    end
+
+    def handle_unverified_user(options)
+      respond_to do |format|
+        format.any(:html, :mobile) do
+          flash[:notice] = options[:notice]
+          redirect_to options[:redirect]
         end
+        format.json { render json: options[:api_notice], status: 401 }
+        format.xml { render xml: options[:api_notice], status: 401 }
       end
+    end
 
-      def require_user_account
-        verify_user(
-          user:       :any,
-          redirect:   login_users_url,
-          notice:     'You must be logged in to do that',
-          api_notice: 'Authorization required'
-        )
-      end
+    def require_user_account
+      verify_user(
+        user: :any,
+        redirect: login_users_url,
+        notice: "You must be logged in to do that",
+        api_notice: "Authorization required"
+      )
+    end
 
-      def load_session_user
-        if session[:user_id] && session[:persistence_token] && !current_user?
-          begin
-            user = User.find(session[:user_id])
-            if user.persistence_token == session[:persistence_token]
-              set_current_user(user)
-            end
-          rescue ActiveRecord::RecordNotFound
-            # No need to do anything if the record does not exist
+    def load_session_user
+      if session[:user_id] && session[:persistence_token] && !current_user?
+        begin
+          user = User.find(session[:user_id])
+          if user.persistence_token == session[:persistence_token]
+            set_current_user(user)
           end
+        rescue ActiveRecord::RecordNotFound
+          # No need to do anything if the record does not exist
         end
       end
+    end
 
-      def handle_temporary_ban
-        if current_user? && current_user.temporary_banned?
-          logger.info "Authentication failed for user:#{current_user.id} (#{current_user.username}) - temporary ban"
-          flash[:notice] = "You have been banned for #{distance_of_time_in_words(Time.now, current_user.banned_until)}!"
-          deauthenticate!
-        end
+    def handle_temporary_ban
```

Comment:
Assignment Branch Condition size for handle_temporary_ban is too high. [19.1/15]